### PR TITLE
feat: add @cacheable directive (experiment) for use in Force CDN caching

### PR DESCRIPTION
**Pull request by mzikherman**
We want to decorate queries with whether or not they can be cached, so we can opt-in queries for caching (_despite_ a user being logged-in). This is very experimental/first-draft-y!

A GraphQL directive is a great generic way to...decorate(!), and would be extensible to other clients and doesn't require needing to override any Relay/query types (to add 'metadata') or anything esoteric like that.

Really, the only odd thing is that this decorator needs to be supported in the schema (so here in Metaphysics), _but it's not actually for use in Metaphysics - it's more for use by Relay middleware to control caching headers when making the GraphQL request_.

I guess it's not _that_ odd, plus this doesn't actually do anything so we can easily remove it.

### CODE REVIEW

File: src/schema/v2/schema.ts [hunk 1]
```diff
@@ -245,6 +245,15 @@ const OptionalFieldDirective = new GraphQLDirective({
...
 })
 
+const CacheableDirective = new GraphQLDirective({
+  name: "cacheable",
+  locations: [
+    DirectiveLocation.FIELD,
+    DirectiveLocation.FRAGMENT_SPREAD,
+    DirectiveLocation.QUERY,
+  ],
+})
```
Comment (damassi) [inline src/schema/v2/schema.ts:255]: I know this is spike-y but can we still follow existing decorator conventions? src/directives

Comment (mzikherman) [reply src/schema/v2/schema.ts:255]: all our directives are vanilla and defined right here in this file (scroll a few lines above this)

Comment (mzikherman) [reply src/schema/v2/schema.ts:255]: (also thinking this should _only_ apply at the query level, that ensures someone has 'checked' the whole query and tagged it as such)

Comment (damassi) [reply src/schema/v2/schema.ts]: Ah, then that's a bit of tech debt. We should move that definition into https://github.com/artsy/metaphysics/blob/main/src/directives/optionalField/optionalFieldsDirectiveExtension.ts, etc

**Pull request merged**